spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BindableProxyFactory.java

```diff
@@ -176,22 +179,19 @@ private String getNamespacePrefixedChannelName(String name) {
 		return this.channelNamespace + "." + name;
 	}
 
-	private void configureSharedMessageChannel(String name, Class<?> channelType, MessageChannel sharedChannel)
+	private void configureSharedMessageChannel(String name, Class<? extends MessageChannel> channelType, MessageChannel sharedChannel)
```

Comment:
now that no "configuration" happens here, perhaps we should consider pushing the `isAssignableFrom` check up, so that this method is only called when a bridge is necessary... which would also simplify it e.g. `bridgeSharedMessageChannel`. Also, the `name` param is no longer used.